mis_builder_contract/models/mis_contract_line.py

```diff
@@ -0,0 +1,69 @@
+from os.path import dirname, join
+
+from odoo import api, fields, models, tools
+
+
+class ContractLineMIS(models.Model):
+    _inherit = "contract.line"
+
+    repetitions = fields.Integer(compute="_compute_repetitions", store=True)
+    amount = fields.Float(compute="_compute_amount", store=True)
+    account_id = fields.Many2one(
+        "account.account", compute="_compute_account_id", store=True
+    )
+
+    @api.depends("partner_id", "contract_id.contract_type")
+    def _compute_account_id(self):
+        for rec in self:
+            if rec.contract_id.contract_type == "purchase":
+                rec.account_id = rec.partner_id.property_account_payable_id
```

Comment:
Sorry, I told you bad. This should be the income/expense account of the product (which may not be put on the product, but on the product category or company level). I think there's a method for getting this value.